src/main/java/bio/terra/cli/service/WorkspaceManagerService.java

```diff
@@ -190,51 +175,51 @@ public WorkspaceDescriptionList listWorkspaces(int offset, int limit) {
    */
   public WorkspaceDescription createWorkspace(
       @Nullable String displayName, @Nullable String description) {
-    WorkspaceApi workspaceApi = new WorkspaceApi(apiClient);
-    try {
-      // create the Terra workspace object
-      UUID workspaceId = UUID.randomUUID();
-      CreateWorkspaceRequestBody workspaceRequestBody = new CreateWorkspaceRequestBody();
-      workspaceRequestBody.setId(workspaceId);
-      workspaceRequestBody.setStage(WorkspaceStageModel.MC_WORKSPACE);
-      workspaceRequestBody.setSpendProfile("wm-default-spend-profile");
-      workspaceRequestBody.setDisplayName(displayName);
-      workspaceRequestBody.setDescription(description);
-
-      // make the create workspace request
-      HttpUtils.callWithRetries(
-          () -> workspaceApi.createWorkspace(workspaceRequestBody),
-          WorkspaceManagerService::isRetryable);
-
-      // create the Google project that backs the Terra workspace object
-      UUID jobId = UUID.randomUUID();
-      CreateCloudContextRequest cloudContextRequest = new CreateCloudContextRequest();
-      cloudContextRequest.setCloudPlatform(CloudPlatform.GCP);
-      cloudContextRequest.setJobControl(new JobControl().id(jobId.toString()));
-
-      // make the initial create context request
-      HttpUtils.callWithRetries(
-          () -> workspaceApi.createCloudContext(cloudContextRequest, workspaceId),
-          WorkspaceManagerService::isRetryable);
-
-      // poll the result endpoint until the job is no longer RUNNING
-      CreateCloudContextResult createContextResult =
-          HttpUtils.pollWithRetries(
-              () -> workspaceApi.getCreateCloudContextResult(workspaceId, jobId.toString()),
-              (result) -> isDone(result.getJobReport()),
-              WorkspaceManagerService::isRetryable,
-              CREATE_WORKSPACE_MAXIMUM_RETRIES,
-              CREATE_WORKSPACE_DURATION_SLEEP_FOR_RETRY);
-      logger.debug("create workspace context result: {}", createContextResult);
-      throwIfJobNotCompleted(
-          createContextResult.getJobReport(), createContextResult.getErrorReport());
-
-      // call the get workspace endpoint to get the full description object
-      return HttpUtils.callWithRetries(
-          () -> workspaceApi.getWorkspace(workspaceId), WorkspaceManagerService::isRetryable);
-    } catch (ApiException | InterruptedException ex) {
-      throw new SystemException("Error creating a new workspace", ex);
-    }
+    return handleClientExceptions(
+        () -> {
```

Comment:
can you make these big lambdas into helper functions?

Reply:
`handleClientExceptions` encapsulates a `try/catch/finally` block. I could make helper function(s) for the logic in the `catch` and `finally` blocks, but then I'd still need the same `try/catch/finally` pattern in every public method of this class. I'm not sure how to encapsulate a `try/catch/finally` block without using a lambda. But open to suggestions!